common/djangoapps/third_party_auth/identityserver3.py

```diff
@@ -0,0 +1,69 @@
+#pylint: disable=W0223
+"""
+    python-social-auth backend for use with IdentityServer3
+    docs: https://identityserver.github.io/Documentation/docsv2/endpoints/authorization.html
+    docs for adding a new backend to python-social-auth:
+    https://python-social-auth.readthedocs.io/en/latest/backends/implementation.html#oauth
+"""
+from social_core.backends.oauth import BaseOAuth2
+from django.utils.functional import cached_property
+
+
+class IdentityServer3(BaseOAuth2):
+    """
+    An extension of the BaseOAuth2 for use with an IdentityServer3 service.
+    """
+
+    name = "identityServer3"
```

Comment:
can we create a constant for `identityServer3`

Reply:
Hmmm... this is the default flow of oauth social_core backends, so you might not see this variable being used in this class but has a long hierarchical pattern. So I don't think we can change that.

Reply:
I meant create a constant that holds value `identityServer3`, I saw this is hardcoded in you code at another place.